app/controllers/student_requests_controller.rb

```diff
@@ -70,11 +70,15 @@ def adminview
   
   def updaterequestbyadmin
     @student_request = StudentRequest.find params[:id]
-    @student_request.state = params[:state]
-    @student_request.admin_notes = params[:notes_for_myself]
-    @student_request.notes_to_student = params[:notes_for_student]
-    @student_request.save!
-    flash[:notice] = "The request was successfully updated to " + @student_request.state
+      if(!params[:state] == " ")
```

Comment:
Instead of checking params[:state] == " ", you can do a check for this as follows if my above comment seems fine to you 
if(@allstates.contains(params[:state]))